unit_tests/test_tlslite_utils_tripledes.py

```diff
@@ -0,0 +1,430 @@
+# coding=utf-8
+# Copyright (c) 2018, Adam Varga
+#
+# See the LICENSE file for legal information regarding use of this file.
+#
+# This is Known Answers Tests file for python_tripledes,py,
+# which implements 3DES cipher in CBC mode
+#
+# CAVP use 3 Keying Options:
+# KO1: KEY1 != KEY2 != KEY3
+# KO2: (KEY1 = KEY3) != KEY2
+# KO3: KEY1 = KEY2 = KEY3
+
+# compatibility with Python 2.6, for that we need unittest2 package,
+# which is not available on 3.3 or 3.4
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+
+from tlslite.utils.python_tripledes import *
+from tlslite.utils.cryptomath import *
+from tlslite.errors import *
+import warnings
+
+class Test3DES_components(unittest.TestCase):
+    # component functions NOT tested from test vectors
+
+    def test_no_iv(self):
+        with self.assertRaises(ValueError):
+            Python_TripleDES(
+                bytearray(b'\x7c\xa1\x10\x45\x4a\x1a\x6e\x57'
+                          b'\x7c\xa1\x10\x45\x4a\x1a\x6e\x57'
+                          b'\x7c\xa1\x10\x45\x4a\x1a\x6e\x57'))
+
+    def test_bad_iv(self):
```

Comment:
why it's "bad_iv"? `test_too_short_iv`seems more expressive to me

Reply:
I just wanted to test some variant of incorrect iv. I didn't want to implement test_too_short_iv and test_too_long_iv

Reply:
but it does test only the too short, so if it is in the name, it automatically informs the reader that this is the purpose of the test case, what kind of error condition it's testing for